product/views.py

```diff
@@ -58,3 +72,119 @@ def get(self, request, product_id):
+                Q(third_category_id=third_category_id)
+            ).select_related('third_category','second_category','second_category__first_category','brand')
+
+            if sort :
+
+                sort_type = {
+                    '0' : '-created_at',
+                    '1' : '-sales_product',
+                    '2' : '-discount_rate',
+                    '3' : 'price',
+                    '4' : '-price'
+                }
+                sorting =[
+                {   'id' : 0,
+                   'name' : '최신순'},
+                {   'id' : 1,
+                  'name' : '인기순'},
+                {   'id' : 2,
+                  'name' : '할인율순'},
+                {   'id' : 3,
+                  'name' : '가격 낮은순'},
+                {   'id' : 4,
+                  'name' : '가격 높은순'}
+                ]
+
+            if first_category_id:
+                first_category_list =[{
+                    'brand'          : product.brand.name,
+                    'title'          : product.title,
+                    'price'          : product.price,
+                    'discount_rate'  : product.discount_rate,
+                    'main_image_url' : product.main_image_url,
+                    'discount_price' :format(int(round(product.price - (product.price * product.discount_rate),-2)),'.2f')
+                    }for product in products.order_by(sort_type[sort])]
+
+                return JsonResponse({'message:':'SUCCESS', 'first_category_list':first_category_list, 'sorting':sorting}, status=200)
+
+            if second_category_id:
+                second_category_list =[{
+                    'brand'          : product.brand.name,
+                    'title'          : product.title,
+                    'price'          : product.price,
+                    'discount_rate'  : product.discount_rate,
+                    'main_image_url' : product.main_image_url,
+                    'discount_price' :format(int(round(product.price - (product.price * product.discount_rate),-2)),'.2f')
+                    }for product in products.order_by(sort_type[sort])]
+
+                return JsonResponse({'message:':'SUCCESS', 'second_category_list':second_category_list, 'sorting':sorting}, status=200)
+
+            if third_category_id:
+                third_category_list =[{
+                    'brand'          : product.brand.name,
+                    'title'          : product.title,
+                    'price'          : product.price,
+                    'discount_rate'  : product.discount_rate,
+                    'main_image_url' : product.main_image_url,
+                    'discount_price' :format(int(round(product.price - (product.price * product.discount_rate),-2)),'.2f')
+                    }for product in products.order_by(sort_type[sort])]
+
+                return JsonResponse({'message:':'SUCCESS', 'third_category_list':third_category_list, 'sorting':sorting}, status=200)
```

Comment:
위의 Q처리에서 이미 카테고리의 여부를 확인했기때문에 뿌려주는 로직은 이렇게 일일히 하는게 아니라 products 결과 한번만 뿌려주면 카테고리에 따라 뿌려지게 됩니다. 내용도 같죠! 한번만 뿌려주세요.  
```suggestion
                return JsonResponse({'data' : product_list, 'sorting':sorting}, status=200)
```

Reply:
수정하였습니다